crates/fuel-core/src/state/in_memory/transaction.rs

```diff
@@ -41,22 +45,31 @@ use std::{
 };
 
 #[derive(Debug)]
-pub struct MemoryTransactionView {
-    view_layer: MemoryStore,
+pub struct MemoryTransactionView<Description = OnChain>
+where
+    Description: DatabaseDescription,
+{
+    view_layer: MemoryStore<Description>,
     // TODO: Remove `Mutex`.
```

Comment:
Looks like you completed this `TODO`.

Reply:
No, that one we will be able to remove with https://github.com/FuelLabs/fuel-core/issues/1589 =)

Reply:
Oh. Lol. Misread that. NVM. I thought the `Mutex` was gone. Clearly not.